src/robotFactory.js

```diff
@@ -1,13 +1,94 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    const {
+      x = 0,
+      y = 0,
+    } = coords;
```

Comment:
```suggestion
  constructor(name, weight, { x: 0, y: 0 }, chipVersion) {
```
You can destructure it like this